packages/gestalt/src/ComboBoxItem.js

```diff
@@ -0,0 +1,93 @@
+// @flow strict
+import { forwardRef, type Node } from 'react';
+import Text from './Text.js';
+import Icon from './Icon.js';
+import focusStyles from './Focus.css';
+
+export type OptionItemType = {|
+  label: string,
+  subtext?: string,
+  value: string,
+|};
+
+type Props = {|
+  id: string,
+  index: number,
+  isHovered: boolean,
+  isSelected: boolean,
+  label: string,
+  onSelect?: ({|
+    item: OptionItemType,
+    event: SyntheticInputEvent<HTMLInputElement>,
+  |}) => void,
+  setHoveredItemIndex: (number) => void,
+  subtext?: string,
+  value: string,
+|};
+
+const ComboBoxItemWithForwardRef: React$AbstractComponent<Props, ?HTMLElement> = forwardRef<
+  Props,
+  ?HTMLElement,
+>(function OptionItem(
+  { isHovered, id, index, isSelected, label, onSelect, setHoveredItemIndex, subtext, value }: Props,
+  ref,
+): Node {
+  const handleEventPreventDefault = (event) => event.preventDefault();
+
+  const handleOnTap = (event) => onSelect?.({ event, item: { label, value, subtext } });
+
+  const handleOnMouseEnter = () => setHoveredItemIndex(index);
+
+  return (
+    <div
+      aria-selected={isSelected}
+      className={focusStyles.hideOutline}
+      id={`${id}-item-${index}`}
+      onClick={handleOnTap}
+      onKeyPress={handleEventPreventDefault}
+      onMouseDown={handleEventPreventDefault}
+      onMouseEnter={handleOnMouseEnter}
+      ref={isHovered ? ref : null}
+      role="option"
+      style={{ cursor: 'pointer', width: '100%' }}
```

Comment:
Rather than inlining all these styles, can we import from the various CSS files and use those classnames? That will prevent style bloat.

Reply:
Previously, we had all these styles within classNames. 
![image](https://user-images.githubusercontent.com/10593890/146074573-0c9fe1e8-1183-4072-97e1-31ccb20c371e.png)

Reply:
It was one of the steps while simplifying code to try to reduce timing. I'll add back, cos I think it's not doing much benefit in terms of performance.